packages/web-pkg/src/composables/piniaStores/extensionRegistry/types.ts

```diff
@@ -0,0 +1,72 @@
+import { Action } from '../../actions'
+import { SearchProvider, SideBarPanel } from '../../../components'
+import { AppNavigationItem } from '../../../apps'
+import { Item } from '@ownclouders/web-client/src/helpers'
+import { FolderView } from '../../../ui'
+import { Slot } from 'vue'
+import { StringUnionOrAnyString } from '../../../utils'
+
+export type ExtensionType = StringUnionOrAnyString<
+  'action' | 'search' | 'sidebarNav' | 'sidebarPanel' | 'folderView' | 'customComponent'
+>
+
+export type ExtensionScope = StringUnionOrAnyString<'resource' | 'user' | 'group'>
+
+export type BaseExtension = {
+  id: string
+  type: ExtensionType
+  extensionPointIds?: string[]
+  scopes?: ExtensionScope[] // TODO: deprecated
+  userPreference?: {
+    optionLabel?: string
+  }
+}
+
+export interface ActionExtension extends BaseExtension {
+  type: 'action'
+  action: Action
+}
+
+export interface SearchExtension extends BaseExtension {
+  type: 'search'
+  searchProvider: SearchProvider
+}
+
+export interface SidebarNavExtension extends BaseExtension {
+  type: 'sidebarNav'
+  navItem: AppNavigationItem
+}
+
+export interface SidebarPanelExtension<R extends Item, P extends Item, T extends Item>
+  extends BaseExtension {
+  type: 'sidebarPanel'
+  panel: SideBarPanel<R, P, T>
+}
+
+export interface FolderViewExtension extends BaseExtension {
+  type: 'folderView'
+  folderView: FolderView
+}
+
+export interface CustomComponentExtension extends BaseExtension {
+  type: 'customComponent'
+  content: Slot
```

Comment:
is a simple component a Slot as well or do we always need to wrap a component inside a wrapper (render) function? 
As we just pass it through to `<component :is=…/>` it should be valid to just pass in a component. If anything is changed here, docs need to be updated as well.

Reply:
Hmm... would always need to be wrapped. But I could actually widen the type to `content: Slot | Component`. That would make sense and works out of the box... what do you think?

Reply:
Yep, that's actually  what I meant!